File: Baekjoon/Level4/4-03/main.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString();
let n = parseInt(input);

plusCycle(n);

function plusCycle(n) {
    let check = n;
    let count = 1;
    while (1) {
        n = calCycle(n);
        if (check == n) break;
        count++;
    }
    console.log(count);
}

function calCycle(n) {
    let b1 = n % 10;
    n /= 10;
    let b2 = Math.floor(n) % 10;
    if (b1 + b2 >= 10)
        return (b1 * 10) + ((b1 + b2) % 10);
    else
        return (b1 * 10) + (b1 + b2);
}

File: Baekjoon/Level4/4-03/main.dart
import 'dart:io';

int calCycle(int n) {
  int b1 = n % 10;
  n ~/= 10;
  int b2 = n.floor() % 10;
  if (b1 + b2 >= 10)
    return (b1 * 10) + ((b1 + b2) % 10);
  else
    return (b1 * 10) + (b1 + b2);
}

int solution(int n) {
  int answer = 1;
  int check = n;
  while (true) {
    n = calCycle(n);
    if (check == n) break;
    answer++;
  }
  return answer;
}

void main() {
  int n = int.parse(stdin.readLineSync()!);

  print(solution(n));
}
